src/Components/Notifications/NotificationsList.tsx

```diff
@@ -40,10 +40,17 @@ const NotificationsList: React.FC<NotificationsListProps> = ({
   const [currentPaginationType, setCurrentPaginationType] = useState(
     paginationType
   )
+
   const nodes = extractNodes(viewer.notifications).filter(node =>
     shouldDisplayNotification(node)
   )
 
+  useEffect(() => {
+    if (nodes?.length > 0) {
+      setLastSeenNotificationDate(nodes[0].publishedAtAbsolute)
+    }
+  }, [nodes])
```

Comment:
`useEffect` will be called every time the component is updated (since the reference type to `nodes` will change every time). what do you think about this code?

```javascript
const nodes = extractNodes(viewer.notifications).filter(node =>
  shouldDisplayNotification(node)
)
const recentNotification = nodes[0]
const recentNotificationPublishedAt = recentNotification?.publishedAtAbsolute

useEffect(() => {
  if (recentNotificationPublishedAt) {
    setLastSeenNotificationDate(recentNotificationPublishedAt)
  }
}, [recentNotificationPublishedAt])
```

Reply:
IMO we should call `setLastSeenNotificationDate` only when `type === "all"` (discussed this point on the zoom call)

Reply:
done, thanks!